plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/PomTest.java

```diff
@@ -0,0 +1,194 @@
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.google.cloud.tools.eclipse.appengine.libraries.model.Library;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.LibraryFile;
+import com.google.cloud.tools.eclipse.appengine.libraries.model.MavenCoordinates;
+import com.google.cloud.tools.eclipse.test.util.project.TestProjectCreator;
+
+public class PomTest {
+  
+  @Rule public final TestProjectCreator projectCreator = new TestProjectCreator();
+
+  // todo we're doing enough of this we should import or write some utilities
+  private static final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+  
+  @BeforeClass 
+  public static void configureParser() {
+    factory.setNamespaceAware(true);
+  }
+  
+  private Pom pom;
+  private IFile pomFile;
+  
+  @Before
+  public void setUp() throws SAXException, IOException, CoreException {
+    IProject project = projectCreator.getProject();
+    pomFile = project.getFile("pom.xml");
+    InputStream in = new FileInputStream(
+       Paths.get("testdata/testpom.xml").toAbsolutePath().toFile());
+    pomFile.create(in, IFile.FORCE, null);
+   
+    pom = Pom.parse(pomFile);
+  }
+  
+  @Test
+  public void testAddDependencies() 
+      throws CoreException, ParserConfigurationException, IOException, SAXException {
+    
+    MavenCoordinates coordinates0 = new MavenCoordinates("com.example.group0", "artifact0");
+    coordinates0.setVersion("1.2.3");
+    LibraryFile file0 = new LibraryFile(coordinates0);
+    List<LibraryFile> list0 = new ArrayList<>();
+    list0.add(file0);
+    
+    List<LibraryFile> list1 = new ArrayList<>();
+    LibraryFile file1 = new LibraryFile(new MavenCoordinates("com.example.group1", "artifact1"));
+    list1.add(file1);
+    
+    List<LibraryFile> list2 = new ArrayList<>();
+    LibraryFile file2 = new LibraryFile(new MavenCoordinates("com.example.group2", "artifact2"));
+    LibraryFile file3 = new LibraryFile(new MavenCoordinates("com.example.group3", "artifact3"));
+    list2.add(file2);
+    list2.add(file3);
+    
+    Library library0 = new Library("id0", list0);
+    Library library1 = new Library("id1", list1);
+    Library library2 = new Library("id2", list2);
+    List<Library> libraries = new ArrayList<>();
+    libraries.add(library0);
+    libraries.add(library1);
+    libraries.add(library2);
+    
+    pom.addDependencies(libraries);
+    
+    InputStream contents = pomFile.getContents();
+    Document actual = parse(contents);
+    
+    NodeList dependencies = actual.getElementsByTagName("dependencies");
+    Assert.assertEquals(1, dependencies.getLength());
+    NodeList children = ((Element) dependencies.item(0)).getElementsByTagName("dependency");
+    
+    Assert.assertEquals(4, children.getLength());
+    
+    Element child0 = (Element) children.item(0);
+    Element groupId = getOnlyChild(child0, "groupId");
```

Comment:
We could extract group ID, artifact ID and version verification into a helper method.